crates/ironrdp-rdpdr-native/src/nix/simple.rs

```diff
@@ -0,0 +1,765 @@
+                // TODO
+                Ok(Vec::new())
+            }
+            ServerDriveIoRequest::ServerDriveQueryDirectoryRequest(req_inner) => query_directory(self, req_inner),
+            ServerDriveIoRequest::ServerDriveQueryInformationRequest(req_inner) => query_information(self, req_inner),
+            ServerDriveIoRequest::ServerDriveQueryVolumeInformationRequest(req_inner) => {
+                query_volume_information(self, req_inner)
+            }
+            ServerDriveIoRequest::ServerDriveSetInformationRequest(req_inner) => set_information(self, req_inner),
+            ServerDriveIoRequest::DeviceControlRequest(req_inner) => Ok(vec![SvcMessage::from(
+                RdpdrPdu::DeviceControlResponse(DeviceControlResponse {
+                    device_io_reply: DeviceIoResponse::new(req_inner.header, NtStatus::SUCCESS),
+                    output_buffer: None,
+                }),
+            )]),
+            ServerDriveIoRequest::ServerDriveLockControlRequest(_) => {
+                // TODO
+                Ok(Vec::new())
+            }
+        }
+    }
+}
+
+pub(crate) fn write_device(
+    backend: &mut SimpleRdpdrBackend,
+    req_inner: DeviceWriteRequest,
+) -> PduResult<Vec<SvcMessage>> {
+    let write_inner = |file: &mut std::fs::File, offset, write_data| -> std::io::Result<usize> {
+        let sf = SeekFrom::Start(offset);
+        file.seek(sf)?;
+        let length = file.write(write_data)?;
+        file.flush()?;
+        Ok(length)
+    };
+    process_dependent_file(
+        backend,
+        req_inner.device_io_request,
+        |request| {
+            let res = RdpdrPdu::DeviceWriteResponse(DeviceWriteResponse {
+                device_io_reply: DeviceIoResponse::new(request, NtStatus::NO_SUCH_FILE),
+                length: 0_u32,
+            });
+            Ok(vec![SvcMessage::from(res)])
+        },
+        |file, request| match write_inner(file, req_inner.offset, &req_inner.write_data) {
+            Ok(length) => {
+                if length == req_inner.write_data.len() {
+                    Ok(vec![SvcMessage::from(RdpdrPdu::DeviceWriteResponse(
+                        DeviceWriteResponse {
+                            device_io_reply: DeviceIoResponse::new(request, NtStatus::SUCCESS),
+                            length: req_inner.write_data.len() as u32,
+                        },
+                    ))])
+                } else {
+                    warn!("write len:{} is not equal to {}", length, req_inner.write_data.len());
+                    let res = RdpdrPdu::DeviceWriteResponse(DeviceWriteResponse {
+                        device_io_reply: DeviceIoResponse::new(request, NtStatus::UNSUCCESSFUL),
+                        length: 0_u32,
+                    });
+                    Ok(vec![SvcMessage::from(res)])
+                }
+            }
+            Err(e) => {
+                warn!("write error:{:?}", e);
```

Comment:
style: See [this section](https://github.com/Devolutions/IronRDP/blob/abc0ee0617fcdcda3f0ace3ffd292251979d1910/STYLE.md#logging)
```suggestion
            Err(error) => {
                warn!(%error, "Write error");
```

Reply:
ok, i modify according to our style